resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/ApplicationMaster.scala

```diff
@@ -90,6 +90,18 @@ private[spark] class ApplicationMaster(
   @volatile private var reporterThread: Thread = _
   @volatile private var allocator: YarnAllocator = _
 
+  private val userClassLoader = {
+    val classpath = Client.getUserClasspath(sparkConf)
+    val urls = classpath.map { entry =>
+      new URL("file:" + new File(entry.getPath()).getAbsolutePath())
+    }
+    if (Client.isUserClassPathFirst(sparkConf, isDriver = true)) {
```

Comment:
Now `spark.driver.userClassPathFirst` will apply to the client AM, is that what you want? Shouldn't it be a separate option?